core/os/os2/errors.odin

```diff
@@ -22,9 +26,15 @@ General_Error :: enum u32 {
 
 Platform_Error :: enum i32 {None=0}
 
+Read_Error :: enum u32 {
+	None,
+	Broken_Pipe,
+}
```

Comment:
Why is this a separate error type?

Reply:
It seems unfit to be `io.Error`, because that one deals with streams in general, also unfit for `General_Error`, because it can only occur during one of the two operations (read, write) and only for pipes, and it's not allocator or some generic platform error. By exclusion I deduced it must live in its own separate type, though now realising `Read_Error` is a bit unfitting. I can't think of a better name as of now.

Is it okay if it lives in`General_Error`?

Reply:
I'd put this in `General_Error` regardless. Yes it can only happen in two cases, but people will want to handle it "general"ly.

Reply:
Done.